src/pages/ExamsPage/utils.js

```diff
@@ -18,3 +18,24 @@ const examAttemptStatusLabels = {
 };
 
 export const getMessageLabelForStatus = (status) => examAttemptStatusLabels[status];
+
+export const createAllowanceData = (form) => {
+  const data = [];
+  const users = form.users.split(',').map((item) => item.trim());
+  for (let i = 0; i < users.length; i++) {
+    const user = users[i];
+    for (let j = 0; j < form.exams.length; j++) {
+      const exam = form.exams[j];
+      let additionalTime;
+      if (form['allowance-type'] === 'time-multiplier') {
+        additionalTime = (exam['time-limit-mins'] * form['additional-time-multiplier']) - exam['time-limit-mins'];
+      } else {
+        additionalTime = form['additional-time-minutes'];
+      }
+
+      const userKey = user.includes('@') ? 'email' : 'username';
```

Comment:
I'm not sure if I should include a more strict check for email vs username.

Reply:
Can you explain why we need to differentiate email or username here? Only for display purposes?

Reply:
Seems practical enough for me.

Reply:
The backend accepts either `email` or `username` for the user identifier, so we need to specify which identifier we are sending from the frontend.